src/types/params.ts

```diff
@@ -149,6 +177,6 @@ export type ExtractParamType<TParam extends Param> =
           : string
 
 type RemoveLeadingQuestionMark<T extends PropertyKey> = T extends `?${infer TRest extends string}` ? TRest : T
-export type RemoveLeadingQuestionMarkFromKeys<T extends Record<string, unknown>> = {
+export type RemoveLeadingQuestionMarkFromKeys<T> = {
```

Comment:
Why remove the extends? This type won't work all `unknown` types which is essentially what this is now right?

Reply:
this was part of testing, I'll put it back but also this type will be removed [here](https://github.com/kitbagjs/router/pull/464) FYI